File: MoonHyuk/04QUADTREE/quadtree.py
testCase = int(input())

def function(string):
    length = 1
    if string[0] == 'x':
        list = []
        for i in range(1, 5):

            list.append(function(string[length:]))
            length += len(list[i-1])

        list[0], list[2] = list[2], list[0]
        list[1], list[3] = list[3], list[1]

        return "x" + "".join(list)

    else:
        return string[0]


for i in range(0, testCase):
    print(function(input()))

File: nicholaskim/04QUADTREE/quadtree.rb
class TestCase
  def initialize(quadtree)
    @quadtree = quadtree.gsub("\n","").split('')
  end

  # Check for exceptions and pre-process the Array
  def process
    if @quadtree.first() == "w" or @quadtree.first() == "b"
      return @quadtree.flatten.join("")
    else
      @quadtree.shift()
      return recurisve.flatten.join("")
    end
  end

  # Recurisve function for tree iteration
  def recurisve
    list = []
    (1..4).each do |i|
      e = @quadtree.shift()
      e = recurisve if e == 'x'
      list << e
    end
    return reverse(list)
  end

  # Returns a reversed array
  def reverse(list)
    return ["x"] + list.pop(2) + list
  end
end

input = $stdin.readlines
number_of_test_cases = input.shift.to_i
output = []

(1..number_of_test_cases).each do |index|
  test_case_input = input.shift()
  t = TestCase.new(test_case_input)
  puts t.process
end
